### ISSUE TITLE
[Due for payment 2026-02-23] Sreen Reader: Settings: Button of Status is announced without selected emoji or not selected status

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77365]
## Action Performed:
Precondition: The User should be logged in

1. Using Android TalkBack or iOS VoiceOver, open the Expensify app
2. Swipe to the 'Account' button at the bottom right corner of the screen and double-tap to activate
3. Swipe to the 'Status' button at the top right corner of the screen
4. Note that the emoji is not selected
5. Note what is announced.

## Expected Result:
The button should be announced with a descriptive name, e.g.:
- 'Status, emoji not selected'
- 'Status, <emoji>'

## Actual Result:
The button of Status is announced without a selected emoji or not unselected state.

## Area issue was found in:
Settings

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure the accessibilityLabel is dynamically updated to include both the status text and the current state, e.g.:
- 'Status, emoji not selected'
- 'Status, <emoji>'
https://reactnative.dev/docs/accessibility#accessibility-properties

## Environment(s)
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos
![Image](https://github.com/user-attachments/assets/e2ad4481-4b08-4cae-b163-84c1e3d3c081)

https://github.com/user-attachments/assets/1df4be27-a6a6-47d7-a774-7e4f0f401ec3

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006019
**Issue reported by:** NA
**Slack conversation:** NA

**Comment by rushatgabhane** [comment]
fixed https://github.com/Expensify/App/issues/79233
